core/src/OpenACC/Kokkos_OpenACC_ParallelScan_Range.hpp

```diff
@@ -0,0 +1,261 @@
+    element_values [0:2 * chunk_size])                                    \
+    present(functor, offset_values, final_reducer) copyin(m_result_total) \
+        async(async_arg)
+    for (IndexType team_id = 0; team_id < n_chunks; ++team_id) {
+      IndexType current_step = 0;
+      IndexType next_step    = 1;
+      IndexType temp;
+#pragma acc loop vector
+      for (IndexType thread_id = 0; thread_id < chunk_size; ++thread_id) {
+        const IndexType local_offset = team_id * chunk_size;
+        const IndexType idx          = local_offset + thread_id;
+        ValueType update;
+        final_reducer.init(&update);
+        if (thread_id == 0) {
+          final_reducer.join(&update, &offset_values(team_id));
+        }
+        if ((idx > 0) && (idx < N)) functor(idx - 1, update, false);
+        element_values[thread_id] = update;
+      }
+      for (IndexType step_size = 1; step_size < chunk_size; step_size *= 2) {
+#pragma acc loop vector
+        for (IndexType thread_id = 0; thread_id < chunk_size; ++thread_id) {
+          if (thread_id < step_size) {
+            element_values[next_step * chunk_size + thread_id] =
+                element_values[current_step * chunk_size + thread_id];
+          } else {
+            ValueType localValue =
+                element_values[current_step * chunk_size + thread_id];
+            final_reducer.join(&localValue,
+                               &element_values[current_step * chunk_size +
+                                               thread_id - step_size]);
+            element_values[next_step * chunk_size + thread_id] = localValue;
+          }
+        }
+        temp         = current_step;
+        current_step = next_step;
+        next_step    = temp;
+      }
+#pragma acc loop vector
+      for (IndexType thread_id = 0; thread_id < chunk_size; ++thread_id) {
+        const IndexType local_offset = team_id * chunk_size;
+        const IndexType idx          = local_offset + thread_id;
+        ValueType update =
+            element_values[current_step * chunk_size + thread_id];
+        if (idx < N) functor(idx, update, true);
+        if (idx == N - 1) {
+          if (m_result_ptr_device_accessible) {
+            *m_result_ptr = update;
+          } else {
+            m_result_total() = update;
+          }
+        }
+      }
+    }
+    if (!m_result_ptr_device_accessible && m_result_ptr != nullptr) {
+      // size_t size = Analysis::value_size(m_functor);
+      DeepCopy<HostSpace, Kokkos::Experimental::OpenACCSpace,
+               Kokkos::Experimental::OpenACC>(m_policy.space(), m_result_ptr,
+                                              m_result_total.data(),
+                                              sizeof(ValueType));
+    }
+
+#pragma acc exit data delete (functor, chunk_values, offset_values, \
+                              final_reducer)async(async_arg)
+    acc_wait(async_arg);
```

Comment:
We need this synchronization since the parallel scan implementation uses several local variables (`chunk_values`, `offset_values`, `m_result_total`, `element_values_owner`) whose lifespan is the enclosing function (`OpenACCParallelScanRangePolicy`). All asynchronous executions here should finish before exiting the enclosing function.